test/lightweight_test_test.cpp

```diff
@@ -65,10 +65,15 @@ int main()
     BOOST_TEST_EQ( ++x, ++y );
     BOOST_TEST_EQ( x++, y++ );
 
+    const char* s1 = "abc"; // make sure addresses are different
+    const char* s2 = "abc"; // make sure addresses are different
```

Comment:
This does not guarantee that `s1` and `s2` are not equal. The compiler may merge equal string literals.